### PR TITLE
[#5] [Integrate] As a user, I can make a shortened link

### PR DESCRIPTION
**Pull request by blyscuit**
#5 

## What happened 👀

Add shorten link api to the screen ShortenLinkViewController.
 
## Insight 📝

Add some extensions.

Add Factory.

Put ShortenLinkViewController on PopupController.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/6356137/142807935-7d8448aa-b286-41e2-8036-6d06545c3b9a.mp4

### CODE REVIEW

File: ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+protocol ShortenLinkViewModelOutput {
+
+    var linkShortenSuccess: Driver<Bool> { get }
+}
+
+protocol ShortenLinkViewModelInput {
+
+    func shortenLink(link: String)
```
Comment (phongvhd93) [inline ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift:25]: You can bind the ` shortenButton.rx.tap` event directly if you use `PublishRelay<String>`

File: ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    }
+                    .asDriver(onErrorJustReturn: false)
+            }
+            .asDriver(onErrorJustReturn: false)
+    }
+}
+
+extension ShortenLinkViewModel: ShortenLinkViewModelInput {
+
+    func shortenLink(link: String) {
```
Comment (phongvhd93) [inline ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift:62]: You won't need this and the `shortenLinkTrigger` if the input of ViewModel is PublishRelay

File: ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift (hunk 1 above)
Comment (blyscuit) [reply ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift:25]: This binding is the same way as our MVVM demo. The benefit is that the input is one way binding where output is another one way binding hence making the viewModel easier to read through.

**Review by phongvhd93 — APPROVED** [review]

**Review by vnntsu — APPROVED** [review]

**Comment by blyscuit** [comment]
I rebased this, please check again.

**Pull request merged**